event_routing_backends/models.py

```diff
@@ -240,6 +245,33 @@ def _match_event_for_host(self, original_event, host_config):
             bool
         """
         for key, value in host_config.get('match_params', {}).items():
-            if get_value_from_dotted_path(original_event, key) != value:
+            original_event_value = get_value_from_dotted_path(original_event, key)
+            if isinstance(value, list):
+                matched = False
+                for value_item in value:
+                    if self._is_match(value_item, original_event_value):
+                        matched = True
+                if not matched:
+                    return False
+            elif not self._is_match(value, original_event_value):
                 return False
         return True
+
+    def _is_match(self, regex_exp, value_str):
+        """
+        Return True if the `regex expression` matches the `value_str`.
+
+        Arguments:
+            regex_exp    (str):     regex expression string
+            value_str       (str):     value string in which we need to compare regex expression
+
+        Returns:
+            bool
+        """
+        try:
+            return bool(re.compile(str(regex_exp))) and re.search(regex_exp, value_str)
+        except TypeError as err:
+            logger.info(
+                        'Invalid regex %s with error: %s', regex_exp, err
+                    )
+            return value_str == regex_exp
```

Comment:
In case of invalid regex `_is_match` should return False

Reply:
Done